### PR TITLE
feat: consensus glossary

### LINKED ISSUE #12307: Consensus Glossary
Create a markdown file in the documentation for writing down definitions for consensus terms.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #12307 

Although this does not define all terms, it gives us a good starting point.

### CODE REVIEW

File: platform-sdk/docs/core/consensus-glossary.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+If the event's claimed parents cause the event to violate the invariant that birth rounds never decrease, then the
+event is immediately discarded as invalid.
+
+The reason why the word "claimed" is used is because it's possible for a malicious node to lie about the existence
+of a parent that may not exist. The fact that a claimed parent birth round may not be correct is not a problem though.
+
+- if an attacker claims to have a parent that doesn't exist, we hold the child event in the orphan buffer until
+  the non-existent parent is ancient
+- if the attacker claims a parent has the wrong birth round and the parent exists, we detect the discrepancy and do not
```
Comment (alittley) [inline platform-sdk/docs/core/consensus-glossary.md:36]: I think the case of an event claiming an incorrect birth round for a parent actually gets resolved before the linking phase, by the orphan buffer.
  
  Birth round is part of the `EventDescriptor`, so if `eventZ` says "my parent is `eventX`", but modifies the birth round of `eventX` -> `eventX_tampered`, then the `OrphanBuffer` will not be able to find the claimed parent, and `eventZ` will sit in the `OrphanBuffer` until `eventX_tampered` goes ancient

We do have logic in the `InOrderLinker` for if the birth round doesn't match. But I haven't been able to think of a way that such a mismatch could actually occur.

File: platform-sdk/docs/core/consensus-glossary.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+An event is considered to be future event if its birth round is greater than the
+[pending round](#pending-consensus-round).
+
+When choosing events to use as parents for a new self event, future events are never selected. This is required to
+maintain the [invariant that birth rounds never decrease](#invariant-birth-rounds-never-decrease).
+
+### Unverifiable Future Event
+
+A future event is considered to be unverifiable future event if we do not yet know what the roster is going to be
```
Comment (alittley) [inline platform-sdk/docs/core/consensus-glossary.md:81]: "A future event is considered to be unverifiable future event"

Should be either "is considered to be *an* unverifiable future event", or "is considered to be unverifiable"

File: platform-sdk/docs/core/consensus-glossary.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+The round that we are currently working on.  Equal to the [latest round](#latest-consensus-round) plus one.
+
+## Rounds Non-Ancient
+
+A configuration parameter for the network. Historically on the Hedera network, this value has been set to 26.
+
+## Preconsensus Generation
+
+The preconsensus generation of an event is defined as the maximum generation of all of that event's parents plus 1.
+If an event's has an ancient parent, then the that parent is ignored for the purpose of this calculation.
```
Comment (alittley) [inline platform-sdk/docs/core/consensus-glossary.md:134]: "If an event's" -> "If an event"

Comment (alittley) [inline platform-sdk/docs/core/consensus-glossary.md:134]: "then the that parent" -> "then that parent"

File: platform-sdk/docs/core/consensus-glossary.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+A famous witness is unique if there are not two famous witnesses in the same round that are created by the same node.
+
+### Invariant: Prerequisite Unique Famous Witnesses
+
+In order to compute consensus for round R, all unique famous witnesses for round R-1 must be in the hashgraph.
+
+## Witnesses
+
+The first event created by a node in a particular [round](#round) is considered to be a witness. As long as a node is
```
Comment (alittley) [inline platform-sdk/docs/core/consensus-glossary.md:163]: Did you mean to say "As long as a node is *NOT* branching"?

**Review by kfa-aguda — APPROVED** [review]

**Comment by cody-littley** [comment]
Closing this for now

**Pull request closed without merging**